### PR TITLE
Link the API docs on waitFor() to the docs for runUnsynchronized()

### LINKED ISSUE #51971: Update waitFor() documentation
link the API docs on `waitFor()` to the docs for `runUnsynchronized()` (https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/runUnsynchronized.html), which has much better guidance.

Noted from guidance on https://buganizer.corp.google.com/issues/150285048 (internal bug).

cc @zanderso

### PR DESCRIPTION
**Pull request by darshankawar**
## Description

This PR adds a reference link of Flutter Driver's `waitFor()` method in `runUnSynchronized()` method doc by explaining how the two methods relate.

## Related Issues

Fixes [51971](https://github.com/flutter/flutter/issues/51971)

## Tests

This does not need tests, as it is purely a documentation PR.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by darshankawar** [comment]
@googlebot I consent

**Comment by DK15** [comment]
@googlebot I consent

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 1]
```diff
@@ -204,11 +204,18 @@ abstract class FlutterDriver {
...
+  ///
+  /// See also:
+  ///
+  /// * [FlutterDriver.runUnsynchronized], which will execute an action
+  /// With frame sync disabled and while the app under test is undergoing
+  /// A transition.
   Future<void> waitFor(SerializableFinder finder, { Duration timeout }) async {
     await sendCommand(WaitFor(finder, timeout: timeout));
   }
-
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:211]: Keep this whitespace?

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 2]
```diff
@@ -684,6 +691,7 @@ abstract class FlutterDriver {
   /// With frame sync disabled, it's the responsibility of the test author to
   /// ensure that no action is performed while the app is undergoing a
   /// transition to avoid flakiness.
+  ///
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:694]: remove this?

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 3]
```diff
@@ -204,11 +204,18 @@ abstract class FlutterDriver {
     await sendCommand(Tap(finder, timeout: timeout));
   }
 
-  /// Waits until [finder] locates the target.
+  /// Waits until [finder] locates the target. [finder] will wait until
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:207]: First paragraph should be a single sentence.

Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:207]: Sentences in docs should be grammatically correct, i.e. start with a capital letter and don't have random capital letters in the middle of it.

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 1 above)
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: fixed

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 2 above)
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: done

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 3 above)
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: fixed

Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: fixed

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 4]
```diff
@@ -205,6 +205,14 @@ abstract class FlutterDriver {
   }
 
   /// Waits until [finder] locates the target.
+  /// The [finder] will wait until There is no pending frame scheduled
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:208]: nit: blank line before this one

Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:208]: There -> there

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 5]
```diff
@@ -205,6 +205,14 @@ abstract class FlutterDriver {
...
+  /// in the app under test Before executing an action.
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:209]: Before -> before

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 6]
```diff
@@ -205,6 +205,14 @@ abstract class FlutterDriver {
...
+  ///
+  /// See also:
+  ///
+  /// * [FlutterDriver.runUnsynchronized], which will execute an action
+  /// with frame sync disabled and while the app under test is undergoing
+  /// a transition.
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:215]: indent these two lines by two spaces.

Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: done

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 5 above)
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: fixed.

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 4 above)
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: fixed

Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: added

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 7]
```diff
@@ -205,6 +205,15 @@ abstract class FlutterDriver {
...
   /// Waits until [finder] locates the target.
+  ///
+  /// The [finder] will wait until there is no pending frame scheduled
+  /// in the app under test before executing an action.
+  ///
+  /// See also:
+  ///
+  ///  * [FlutterDriver.runUnsynchronized], which will execute an action
+  ///  with frame sync disabled and while the app under test is undergoing
+  ///  a transition.
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:216]:
```suggestion
  ///  * [FlutterDriver.runUnsynchronized], which will execute an action
  ///    with frame sync disabled even while frames are pending.
```

Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart:216]: @goderbauer I've made these changes. Will push update.

Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: @goderbauer  pushed this change and ready for a look.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**